src/sage/rings/polynomial/multi_polynomial_ideal.py

```diff
@@ -3919,10 +3918,12 @@ def __hash__(self):
         return 0
 
     @cached_method
-    def gens(self):
+    def gens(self):  # -> PolynomialSequence
```

Comment:
Same question here

Reply:
At least here its relatively clear that it's returning `PolynomialSequence_generic` or not?

Reply:
I have used `Sequence_generic` here